modules/tinymce/src/plugins/codesample/test/ts/browser/CodeSampleSelectionTest.ts

```diff
@@ -1,59 +1,46 @@
+import { SugarBody } from '@ephox/sugar';
 
-    Pipeline.async({},
-      [
-        Log.stepsAsStep(
-          'TBA',
-          'CodeSample: TBA-Open the dialog and check it has the right initial values. ' +
-          'Set the codesample content, submit and check the editor content changes correctly. ' +
-          'Double-click on the editor and check if the dialog opens with the correct language and content.', [
-            TestUtils.sOpenDialogAndAssertInitial(editor, docBody, 'markup', ''),
-            TestUtils.sSetTextareaContent(markupContent),
-            TestUtils.sSubmitDialog(docBody),
-            TestUtils.sAssertEditorContents(editorBody, 'markup', markupContent, 'pre.language-markup'),
-            Step.sync(() => {
-              const pre = editor.getBody().querySelector('pre');
-              editor.fire('dblclick', { target: pre });
-            }),
-            UiFinder.sWaitForVisible('Waited for dialog to be visible', docBody, dialogSelector),
-            TestUtils.sAssertCodeSampleDialog('markup', markupContent)
-          ]
-        ),
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/codesample/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+import * as TestUtils from '../module/CodeSampleTestUtils';
 
-        Log.stepsAsStep('TBA', 'CodeSample: Selecting code sample should update button state', [
-          tinyApis.sSetContent('<p>abc</p><pre class="language-markup"><code></code></pre>'),
-          tinyApis.sSelect('p', []),
-          tinyApis.sNodeChanged(),
-          UiFinder.sNotExists(docBody, 'button[aria-pressed="true"]'),
-          tinyApis.sSelect('pre.language-markup', []),
-          tinyApis.sNodeChanged(),
-          UiFinder.sExists(docBody, 'button[aria-pressed="true"]')
-        ])
-      ]
-      , onSuccess, onFailure);
-  }, {
+describe('browser.tinymce.plugins.codesample.CodeSampleSelectionTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: 'codesample',
-    theme: 'silver',
     toolbar: 'codesample',
     base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
+  }, [ Plugin, Theme ]);
+
+  const dialogSelector = 'div.tox-dialog';
+  const markupContent = '<p>hello world</p>';
+
+  it('TBA: Open the dialog and check it has the right initial values. ' +
+    'Set the codesample content, submit and check the editor content changes correctly. ' +
+    'Double-click on the editor and check if the dialog opens with the correct language and content.', async () => {
+    const editor = hook.editor();
+    await TestUtils.pOpenDialogAndAssertInitial(editor, 'markup', '');
+    TestUtils.setTextareaContent(markupContent);
+    await TestUtils.pSubmitDialog(editor);
+    await TestUtils.pAssertEditorContents(TinyDom.body(editor), 'markup', markupContent, 'pre.language-markup');
+    const pre = editor.getBody().querySelector('pre');
+    editor.fire('dblclick', { target: pre } as unknown as MouseEvent);
```

Comment:
You can take out the `as unknown` by changing the line above to `const pre: EventTarget`

Reply:
I don't want to do that, as it's down casting the type and would cause issues if you had to use `pre` for anything else.

Reply:
I really wish TS would just pick this up, given `Node` extends `EventTarget` :cry:

Reply:
So it turns out that whatever relationship (has some of the fields, or something?) that typescript is looking for between {expression} and {type to cast expression as} can't handle any of the fields extending the type that it expected for that field. Seems like a bit of a flaw on their end, but not much we can do.